pkg/lib/_internal/dnf5daemon.ts

```diff
@@ -297,28 +340,299 @@ export class Dnf5DaemonManager implements PackageManager {
+            case 'transaction_elem_progress': {
+                const [, _last_name, processed,] = args as [string, string, number, number];
+                last_progress = processed / total_packages * 100;
+                break;
+            }
+            }
+
+            if (progress_cb) {
+                progress_cb({
+                    waiting: false,
+                    percentage: last_progress,
+                    cancel: null,
+                });
+            }
+        }
+
+        await this.with_session(async (session) => {
+            await call(session, "org.rpm.dnf.v0.rpm.Rpm", "install", [pkgnames, {}]);
+            const [_transaction_items, result] = await call(session, "org.rpm.dnf.v0.Goal", "resolve", [{}]) as InstallResolveResult;
+            if (result !== 0) {
+                const [problems] = await call(session, "org.rpm.dnf.v0.Goal", "get_transaction_problems", []) as TransactionProblem[][];
+                if (problems.every((p: TransactionProblem) => p.problem.v == GoalProblem.ALREADY_INSTALLED)) {
+                    await call(session, "org.rpm.dnf.v0.Goal", "reset", []);
+                    return;
+                }
+
+                const [problem] = await call(session, "org.rpm.dnf.v0.Goal", "get_transaction_problems_string", []);
+                throw new ResolveError(`Resolving install failed with result=${result}. ${problem}`);
+            }
+            await call(session, "org.rpm.dnf.v0.Goal", "do_transaction", [{}]);
+        }, signal_emitted);
+    }
+
+    async remove_packages(pkgnames: string[], progress_cb?: ProgressCB): Promise<void> {
+        let last_progress = 0;
+        let total_packages: number;
+
+        function signal_emitted(_path: string, _iface: string, signal: string, args: unknown[]) {
+            switch (signal) {
+            case 'transaction_before_begin':
+                [, total_packages] = args as [string, number];
+                break;
+            case 'transaction_elem_progress': {
+                const [, _last_name, processed,] = args as [string, string, number, number];
+                last_progress = processed / total_packages * 100;
+                break;
+            }
+            }
+
+            if (progress_cb) {
+                progress_cb({
+                    waiting: false,
+                    percentage: last_progress,
+                    cancel: null,
+                });
+            }
+        }
+
+        await this.with_session(async (session) => {
+            await call(session, "org.rpm.dnf.v0.rpm.Rpm", "remove", [pkgnames, {}]);
+            const [_transaction_items, result] = await call(session, "org.rpm.dnf.v0.Goal", "resolve", [{}]) as RemoveResolveResult;
+            if (result !== 0) {
+                const [problem] = await call(session, "org.rpm.dnf.v0.Goal", "get_transaction_problems_string", []);
+                throw new ResolveError(`Resolving remove failed with result=${result}. ${problem}`);
```

Comment:
These 3 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-22611-295a8796-20251126-172456-fedora-43-devel/Coverage/lcov/github-pr.diff.gcov.html)